pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/ExpressionTransformer.java

```diff
@@ -34,33 +33,31 @@
  * regular column for other record transformers.
  */
 public class ExpressionTransformer implements RecordTransformer {
-  private static final Logger LOGGER = LoggerFactory.getLogger(ExpressionTransformer.class);
 
-  private final Map<String, FunctionExpressionEvaluator> _expressionEvaluators = new HashMap<>();
+  private final Map<String, ExpressionEvaluator> _expressionEvaluators = new HashMap<>();
+
+  private final String _timeColumn;
 
   public ExpressionTransformer(Schema schema) {
+    _timeColumn = schema.getTimeColumnName();
     for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
-      if (!fieldSpec.isVirtualColumn()) {
-        String expression = fieldSpec.getTransformFunction();
-        if (expression != null) {
-          try {
-            _expressionEvaluators.put(fieldSpec.getName(), new FunctionExpressionEvaluator(expression));
-          } catch (Exception e) {
-            LOGGER.error("Caught exception while constructing expression evaluator for: {}, skipping", expression, e);
-          }
-        }
+      ExpressionEvaluator expressionEvaluator = ExpressionEvaluatorFactory.getExpressionEvaluator(fieldSpec);
+      if (expressionEvaluator != null) {
+        _expressionEvaluators.put(fieldSpec.getName(), expressionEvaluator);
       }
     }
   }
 
   @Override
   public GenericRow transform(GenericRow record) {
-    for (Map.Entry<String, FunctionExpressionEvaluator> entry : _expressionEvaluators.entrySet()) {
+    for (Map.Entry<String, ExpressionEvaluator> entry : _expressionEvaluators.entrySet()) {
```

Comment:
Shall we consider make this evaluators to be ordered? So we can derived a field based on another derived field?

Reply:
In this iteration, I haven't considered derived fields. It's part of my next steps. Yes, can make this map ordered then. Will also have to include logic in the ExpressionTransformer constructor to come up with the ordering.
The next steps are:
1. Derived fields
2. Custom functions
3. Some default time related functions
4. Use all of the above to migrate to dateTimeFieldSpec
5. Advanced transformations - filter, flatten
6. Pluggable RecordTransformer